src/IB/IBFEMethod.cpp

```diff
@@ -947,14 +947,18 @@ IBFEMethod::spreadForce(const int f_data_idx,
     Pointer<PatchHierarchy<NDIM> > hierarchy = d_use_scratch_hierarchy ? d_scratch_hierarchy : d_hierarchy;
     const int ln = hierarchy->getFinestLevelNumber();
     const auto f_scratch_data_idx = d_active_eulerian_data_cache->getCachedPatchDataIndex(f_data_idx);
+    const auto f_prolong_scratch_data_idx = d_active_eulerian_data_cache->getCachedPatchDataIndex(f_data_idx);
     // zero data.
     Pointer<hier::Variable<NDIM> > f_var;
     VariableDatabase<NDIM>::getDatabase()->mapIndexToVariable(f_data_idx, f_var);
     auto f_active_data_ops = HierarchyDataOpsManager<NDIM>::getManager()->getOperationsDouble(f_var, hierarchy);
-    f_active_data_ops->resetLevels(ln, ln);
+    f_active_data_ops->resetLevels(getCoarsestPatchLevelNumber(), getFinestPatchLevelNumber());
     f_active_data_ops->setToScalar(f_scratch_data_idx,
                                    0.0,
                                    /*interior_only*/ false);
+    f_active_data_ops->setToScalar(f_prolong_scratch_data_idx,
```

Comment:
If this will only show up as the scratch context in a `RefineSchedule`, I don't think you need to zero it out.

Reply:
At first I agreed but when I made this change the tests started to fail (it looks like we end up reading uninitialized data) so we need it for reasons I don't fully understand.

Reply:
Yeah, but the way that you are using it here, I think you _do_ need to zero out --- you are not just using it as scratch data.